checks/branch_protection.go

```diff
@@ -203,11 +429,28 @@ func checkReleaseAndDevBranchProtection(
 	}
 }
 
-// isBranchProtected checks branch protection rules on a Git branch.
-func isBranchProtected(protection *clients.BranchProtectionRule, branch string, dl checker.DetailLogger) int {
-	totalScore := 15
+func testScore(protection *clients.BranchProtectionRule,
+	branch string, dl checker.DetailLogger) (int, error) {
+	var score levelScore
+	score.scores.basic, score.maxes.basic = basicNonAdminProtection(protection, branch, dl)
+	score.scores.adminBasic, score.maxes.adminBasic = adminBasicProtection(protection, branch, dl)
+	score.scores.review, score.maxes.review = nonAdminReviewProtection(protection)
+	score.scores.adminReview, score.maxes.adminReview = adminReviewProtection(protection, branch, dl)
+	score.scores.context, score.maxes.context = nonAdminContextProtection(protection, branch, dl)
+	score.scores.thoroughReview, score.maxes.thoroughReview =
+		nonAdminThoroughReviewProtection(protection, branch, dl)
+	score.scores.adminThoroughReview, score.maxes.adminThoroughReview =
+		adminThoroughReviewProtection(protection, branch, dl)
+
+	return computeScore([]levelScore{score})
+}
```

Comment:
Move this fn to `_test.go` file where it is used.

Reply:
good call! Done. Thanks!